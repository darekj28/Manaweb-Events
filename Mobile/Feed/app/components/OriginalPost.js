import React from 'react';
import { AppRegistry,StyleSheet,Text,View,ListView,TouchableOpacity,TouchableHighlight, TextInput,
        TouchableWithoutFeedback, Alert, Image, Animated} from 'react-native';
const PROFILE_HEIGHT = 50
const PROFILE_WIDTH = 50
export default class OriginalPost extends React.Component {
    constructor() {
        super();
        this.state = {
            show : true
        };
    }
    toggleShow() {
        this.setState({ show : !this.state.show });
    }
	render() {
		var post = this.props.post;
		return (
			<View style={{  flex:1, flexDirection: 'row', justifyContent: 'flex-start',backgroundColor : 'white'}}>
                <View style={{paddingLeft : 8, paddingTop : 8, paddingBottom :8, justifyContent: 'flex-start'}}>
                    {post.avatar =='nissa' && <Image  style={styles.profile_image} source={require('../static/avatars/nissa.png')} />}
                    {post.avatar == 'chandra' && <Image  style={styles.profile_image} source={require('../static/avatars/chandra.png')} />}
                    {post.avatar == 'elspeth' && <Image  style={styles.profile_image} source={require('../static/avatars/elspeth.png')} />}
                    {post.avatar == 'nicol' && <Image  style={styles.profile_image} source={require('../static/avatars/nicol.png')} />}
                    {post.avatar == 'ugin' && <Image  style={styles.profile_image} source={require('../static/avatars/ugin.png')} />}
                    {post.avatar == 'jace' && <Image  style={styles.profile_image} source={require('../static/avatars/jace.png')} />}
                    {post.avatar == 'liliana' && <Image  style={styles.profile_image} source={require('../static/avatars/liliana.png')} />}
                    {post.avatar == 'ajani' && <Image  style={styles.profile_image} source={require('../static/avatars/ajani.png')} />}
                    {post.avatar == 'nahiri' && <Image  style={styles.profile_image} source={require('../static/avatars/nahiri.png')} />}
                    {post.avatar == 'gideon' && <Image  style={styles.profile_image} source={require('../static/avatars/gideon.png')} />}
                </View>
                <View style={{flex : 1,flexDirection : 'column', justifyContent: 'flex-start' }}>
                    <View style={{flexDirection: 'row', justifyContent : 'flex-start', flexWrap : 'wrap', 
                                            paddingLeft : 8, paddingRight : 8, paddingTop : 8}}> 
                        <Text style = {styles.text_name}>
                            {post.name}
                        </Text>
                        <Text style = {styles.text_userID}>
                            @{post.userID}
                        </Text>
                    </View>
                    {this.state.show && <View style={{justifyContent: 'flex-start', padding : 8, paddingBottom : 25, paddingTop : 25 }}>
                        <Text style = {styles.text_message}>
                            {post.postContent}
                        </Text>
                    </View>}
                    <View style={{justifyContent: 'flex-start', padding : 8, paddingTop : 16}}>
                        {!this.state.show && 
                        <TouchableWithoutFeedback onPress={this.toggleShow.bind(this)}><View><Text style = {styles.toggle_message}>
                            Show
                        </Text></View></TouchableWithoutFeedback>}
                        {this.state.show && 
                        <TouchableWithoutFeedback onPress={this.toggleShow.bind(this)}><View><Text style = {styles.toggle_message}>
                            Hide
                        </Text></View></TouchableWithoutFeedback>}
                    </View>
                    <View style={{justifyContent: 'flex-start', padding : 8, paddingLeft : 5}}>
                        <Text style={styles.text_time}>
                            {post.timeString}
                        </Text>
                    </View>
                </View>
            </View>

		);
	}
}
const styles = StyleSheet.create({
    text_name: {
        flex: 0,
        color: '#333333',
        fontSize: 14,
        fontWeight: 'bold',
        textAlignVertical: 'top',
    },
    text_userID: {
        flex: 0,
        fontSize: 14,
        textAlignVertical: 'top',
        color : '#333333',
        paddingLeft : 4
    },
    text_time: {
        flex: 0,
        fontSize: 14,
        textAlignVertical: 'top',
        color : '#333333',
        paddingLeft : 4
    },
    text_message: {
        flex: 0,
        fontSize: 18,
        textAlignVertical: 'top',
        color : '#333333'
    },
    toggle_message : {
        fontSize : 12,
        color : '#90D7ED'
    },
    profile_image: {
        width: PROFILE_WIDTH,
        height: PROFILE_HEIGHT,
        borderRadius: 4
    }
});