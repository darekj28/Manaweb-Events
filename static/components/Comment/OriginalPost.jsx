var React = require('react');
export default class OriginalPost extends React.Component{
	render() {
		return(
			<div id="OriginalPost" className="pull-left">
				<h1> {this.props.original_post['postContent']} </h1> 
			</div>
			);
	}
}