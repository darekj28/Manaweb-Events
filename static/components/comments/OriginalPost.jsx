class OriginalPost extends React.Component{
	constructor(props) {
		super(props);
	}



	render() {
		return(
			<div id="OriginalPost" className="pull-left">
				<h1> {this.props.original_post['postContent']} </h1> 
			</div>
			);
	}
}