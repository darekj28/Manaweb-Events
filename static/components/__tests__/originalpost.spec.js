import OriginalPost from "../OriginalPost.jsx";

describe("OriginalPost", () => {
	it("renders correctly",() => {
		const p = {postContent : "a"};
		const op = renderer.create(
			<OriginalPost original_post={p}/>
			);
		expect(op).toMatchSnapshot();
	});
});