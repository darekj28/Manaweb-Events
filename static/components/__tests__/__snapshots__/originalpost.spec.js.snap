exports[`OriginalPost renders correctly 1`] = `
<div
  className="pull-left"
  id="OriginalPost">
  <h1>
     
    a
     
  </h1>
</div>
`;
